Firestore rules documentation (#187)
mikaelacaron/Basic-Car-Maintenance: Basic-Car-Maintenance/Documentation.docc/FirestoreDetails.md

The alerts collection contains system level alerts that will be visible to all users. The alerts
will only be displayed to the user once.

| Fields     | Firestore Type | Description                                                                            |
| ---------- | -------------- | -------------------------------------------------------------------------------------- |
| actionText | Text           | Text that will appear on the action button                               |
| actionURL  | Text           | Url that the user will be redirected to when the action button is tapped |
| emojiIcon  | Text           | The emoji that will appear on the alert message                                        |
| id         | Text           | Identifies a specific alert                                                            |
| isOn       | Boolean        | Toggles if the alert is active. Only one alert will be active at a time.             |
| message    | Text           | Text content of the alert                                                          |
| title      | Text           | Name given to the alert                                                            |

> ### maintainance_events

Reviewer: ```suggestion
### maintenance_events
```

> Maintainance events collection contains the individual maintainance events that are associated

Reviewer: ```suggestion
Maintenance events collection contains the individual maintenance events that are associated
```

with a specific user.

| Fields    | Firestore Type | Description                                                                               |
| --------- | -------------- | ----------------------------------------------------------------------------------------- |
| id        | DocumentID     | Unique maintainance event identifier that will auto-populate your objects id property |
| userID    | Text           | Identifies the user associated with the maintainance event                                |
| title     | Text           | Customized name given to the maintainance event                                       |

> | date      | Date           | Date that the maintainance event was created                                          |

Reviewer: ```suggestion
| date      | Date           | Date that the maintenance event was recorded                                          |
```

Going to word it this way because technically it's not the date it's created, but whatever date when someone decides to enter it

> | notes     | Text           | Extra details that the user want to provide about the maintainance event                  |

Reviewer: ```suggestion
| notes     | Text           | Extra details about the maintenance event                  |
```

> | vehicleID | Text           | Identifies the vehicle assocated with the maintainance event                              |

Reviewer: ```suggestion
| vehicleID | Text           | Identifies the vehicle associated with the maintenance event                              |
```

typo

### vehicles

The vehicles collection contains all the vehicles associated with a specific user.

| Fields | Firestore Type | Description                                                                    |
| ------ | -------------- | ------------------------------------------------------------------------------ |
| id     | DocumentID     | Unique vehicle identifier that will auto-populate your objects id property |
| userID | Text           | Identifies the user associated with the vehicle                                |
| name | Text | Customized name given to vehicle entry

> | make | Text | Indicates the brand of the vehicle

Reviewer: ```suggestion
| make | Text | Brand of the vehicle
```

## Firestore Collections

### _alerts_

> The alerts collection contains system level alerts that will be visable to all users. The alerts

Reviewer: ```suggestion
The alerts collection contains system level alerts that will be visible to all users. The alerts
```

will only be displayed to the user once.

| Fields     | Firestore Type | Description                                                                            |
| ---------- | -------------- | -------------------------------------------------------------------------------------- |
| actionText | Text           | Indicates the text that will appear on the action button                               |

> | actionURL  | Text           | Indicates the url that the user will be redirected to when the acton button is pressed |

Reviewer: ```suggestion
| actionURL  | Text           | Indicates the url that the user will be redirected to when the acton button is tapped |
```

| emojiIcon  | Text           | The emoji that will appear on the alert message                                        |
| id         | Text           | Identifies a specific alert                                                            |

> | isOn       | Boolean        | Indicates if the alert is active                                                       |

Reviewer: ```suggestion
| isOn       | Boolean        | Indicates if the alert is active. Only one alert will be active at a time.                                                       |
```

| message    | Text           | The text content of the alert                                                          |
| title      | Text           | The name given to the alert                                                            |

> ### _maintainance_events_

Reviewer: No need to make this italic, just make it a regular heading

Maintainance events collection contains the individual maintainance events that are associated
with a specific user.

| Fields    | Firestore Type | Description                                                                               |
| --------- | -------------- | ----------------------------------------------------------------------------------------- |
| id        | DocumentID     | The unique maintainance event identifier that will auto-populate your objects id property |
| userID    | Text           | Identifies the user associated with the maintainance event                                |
| title     | Text           | The customized name given to the maintainance event                                       |
| date      | Date           | The date that the maintainance event was created                                          |
| notes     | Text           | Extra details that the user want to provide about the maintainance event                  |
| vehicleID | Text           | Identifies the vehicle assocated with the maintainance event                              |

> ### _vehicles_

Reviewer: Same for this one about being italic

The vehicles collection contains all the vehicles associated with a specific user.

| Fields | Firestore Type | Description                                                                    |
| ------ | -------------- | ------------------------------------------------------------------------------ |
| id     | DocumentID     | The unique vehicle identifier that will auto-populate your objects id property |
| userID | Text           | Identifies the user associated with the vehicle                                |

| name | Text | Customized name given to vehicle entry
| make | Text | Indicates the brand of the vehicle
| model | Text | Indicates unique name associated with a specific brand
| year | Text | Indicates the year the vehicle was released by the manufacturer

> | color | Text | Indicates the manifacturer color associated with the vehicle

Reviewer: Instead of using the word "indicates" in front of a lot of properties, you can leave the word off (can you change all the properties this affects
